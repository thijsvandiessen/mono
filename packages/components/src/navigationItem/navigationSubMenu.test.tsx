import { describe, expect, it } from 'vitest'
import { render, screen } from '@testing-library/react'
import { NavigationItem } from '../navigationItem/index.js'

describe('NavigationItem', () => {
  it('should render a link', () => {
    const { container } = render(
      <NavigationItem slug={'url'} label={'homepage'} />
    )

    expect(screen.getByRole('link', { name: 'homepage' }))
    expect(container).toMatchSnapshot()
  })

  it('should render a link warning', () => {
    render(<NavigationItem />)

    expect(screen.getByText('No link specified'))
  })

  it('should render a label warning', () => {
    render(<NavigationItem slug={'test'} />)

    expect(screen.getByText('No label specified'))
  })
})
