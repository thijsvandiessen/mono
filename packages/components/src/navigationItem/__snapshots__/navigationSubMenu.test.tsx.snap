// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavigationItem > should render a link 1`] = `
<div>
  <li
    class="_navItem_5ff49f"
  >
    <a
      class="_link_5ff49f text-large"
      href="/url"
    >
      homepage
    </a>
  </li>
</div>
`;
